docs/guides/pagination/offset-based.md

```diff
@@ -0,0 +1,394 @@
+---
+title: Pagination - Offset based
+---
+
+# Implementing Offset Pagination
+
+Make sure to check our introduction to pagination [here](./overview.md)!
+
+Let us implement offset based pagination in GraphQL. By the end of this tutorial, we
+should be able to return a list of users which can be paginated by the client.
+
+```graphql+response
+query getUsers {
+  getUsers(offset: 0, limit: 2) {
+    users {
+      id
+      name
+      occupation
+      age
+    }
+    pageMeta {
+      total
+      pages
+    }
+  }
+}
+---
+{
+  "data": {
+    "getUsers": {
+      "users": [
+        {
+          "id": 1,
+          "name": "Norman Osborn",
+          "occupation": "Founder, Oscorp Industries",
+          "age": 42
+        },
+        {
+          "id": 2,
+          "name": "Peter Parker",
+          "occupation": "Freelance Photographer, The Daily Bugle",
+          "age": 16
+        }
+      ],
+      "pageMeta": {
+          "total": 4,
+          "page": 1,
+          "pages": 2
+      }
+    }
+  }
+}
+```
+
+Let us model our schema like this:
+
+```py
+# example.py
+
+from typing import List
+
+import strawberry
+
+
+@strawberry.type
+class User:
+    name: str = strawberry.field(
+        description="""
+        The name of the user.
+        """
+    )
+
+    occupation: str = strawberry.field(
+        description="""
+        The occupation of the user.
+        """
+    )
+
+    age: int = strawberry.field(
+        description="""
+        The age of the user.
+        """
+    )
+
+
+@strawberry.type
+class PageMeta:
+    total: int = strawberry.field(
+        description="""
+        The total number of items in the dataset.
+        """
+    )
+
+    page: int = strawberry.field(
+        description="""
+        The current page number in the dataset.
+        """
+    )
+
+    pages: int = strawberry.field(
+        description="""
+        The total number of pages in the dataset.
+        """
+    )
+
+
+@strawberry.type
+class UserResponse:
+    users: List[User] = strawberry.field(
+        description="""
+        The list of users.
+        """
+    )
+
+    page_meta: PageMeta = strawberry.field(
+        description="""
+        Metadata to aid in pagination.
+        """
+    )
+
+
+@strawberry.type
+class Query:
+    @strawberry.field(description="Get a list of users.")
+    def get_users(self) -> UserResponse:
+        ...
+
+schema = strawberry.Schema(query=Query)
+
+```
+
+As you can see above, we have modelled our field to return an object type, rather than a list.
```

Comment:
```suggestion
As you can see above, we have modeled our field to return an object type, rather than a list.
```
🇺🇸🦅

Reply:
I didnt know we were following american English lol

Reply:
I'm actually not sure what we're using, lol. I think @patrick91 has been using British English everywhere. I meant to make it clear this was more of a joke than an actual request but I guess I forgot :sweat: .